vsphere/resource_vsphere_host_virtual_switch.go

```diff
@@ -0,0 +1,137 @@
+package vsphere
+
+import (
+	"fmt"
+
+	"context"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi"
+)
+
+func resourceVSphereHostVirtualSwitch() *schema.Resource {
+	s := map[string]*schema.Schema{
+		"name": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The name of the virtual switch.",
+			Required:    true,
+			ForceNew:    true,
+		},
+		"host_system_id": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The managed object ID of the host to set the virtual switch up on.",
+			Required:    true,
+			ForceNew:    true,
+		},
+	}
+	mergeSchema(s, schemaHostVirtualSwitchSpec())
+
+	// Transform any necessary fields in the schema that need to be updated
+	// specifically for this resource.
+	s["active_nics"].Required = true
+	s["standby_nics"].Required = true
+
+	s["teaming_policy"].Default = hostNetworkPolicyNicTeamingPolicyModeLoadbalanceSrcID
+	s["check_beacon"].Default = false
+	s["notify_switches"].Default = true
+	s["failback"].Default = true
+
+	s["allow_promiscuous"].Default = false
+	s["allow_forged_transmits"].Default = true
+	s["allow_mac_changes"].Default = true
+
+	s["shaping_enabled"].Default = false
+
+	return &schema.Resource{
+		Create: resourceVSphereHostVirtualSwitchCreate,
+		Read:   resourceVSphereHostVirtualSwitchRead,
+		Update: resourceVSphereHostVirtualSwitchUpdate,
+		Delete: resourceVSphereHostVirtualSwitchDelete,
+		Schema: s,
+	}
```

Comment:
are we supporting Import for this resource too?

Reply:
Import will be added eventually, but for now it is not going to be supported in this resource (DM'ed you on slack)